#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.5110878622553924 0.8439161594366555 0.16307885484298304 5.524550538289621
  position -2.5554086088793144 1.2156080767975905 3.202376604533808
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0 0 0.23
  rotation 0 0 1 0
  floorSize 2.8 2.8
  floorAppearance PBRAppearance {
    roughness 0.8
    metalness 0
  }
}
Solid {
  translation -1.1 -0.0049 1.45
  rotation 1 0 0 -1.5707996938995747
  scale 0.001 0.001 0.001
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 0.8
        metalness 0
      }
      geometry DEF LINE Mesh {
        url [
          "D:/New/WEBOTS/path_3/z_path/line 2.STL"
        ]
      }
    }
  ]
  boundingObject USE LINE
}
Robot {
  translation -0.766536 0.0148318 0.334123
  rotation -0.001409159980056005 0.9999749858472449 -0.006931229901901547 3.00311
  children [
    Solid {
      translation 0 0.12 0.03
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Cone {
            bottomRadius 0.02
            height 0.07
          }
        }
      ]
      name "solid(8)"
    }
    Solid {
      translation 0 0.15 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.4 0.3 0.2
            metalness 0
          }
          geometry Cylinder {
            height 0.007
            radius 0.03
          }
        }
      ]
      name "solid(6)"
    }
    Solid {
      translation 0 0.157 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.4 0.3 0.2
            metalness 0
          }
          geometry Cylinder {
            height 0.02
            radius 0.022
          }
        }
      ]
      name "solid(7)"
    }
    Solid {
      translation 0 0.05 0
      children [
        Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Sphere {
            radius 0.05
            subdivision 2
          }
        }
      ]
      name "solid(4)"
    }
    Solid {
      translation 0 0.12 0
      children [
        Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Sphere {
            radius 0.033
            subdivision 2
          }
        }
      ]
      name "solid(5)"
    }
    Solid {
      translation -0.015 0.13 0.02
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            metalness 0
          }
          geometry Sphere {
            radius 0.009
            subdivision 2
          }
        }
      ]
      name "solid(9)"
    }
    Solid {
      translation 0.015 0.13 0.02
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            metalness 0
          }
          geometry Sphere {
            radius 0.009
            subdivision 2
          }
        }
      ]
      name "solid(10)"
    }
    DEF RightSensor DistanceSensor {
      translation -0.03 0.0072 0.043
      rotation -0.7071067811865475 0.7071067811865475 0 3
      children [
        Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      name "RightSensor"
      type "infra-red"
    }
    DEF LeftSensor DistanceSensor {
      translation 0.03 0.0072 0.043
      rotation -0.7071067811865475 0.7071067811865475 0 3
      children [
        Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      name "LeftSensor"
      type "infra-red"
    }
    Solid {
      translation 0 -0.007 0.03
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry DEF STEEL_BALL Sphere {
            radius 0.007
          }
        }
      ]
      name "solid(2)"
      boundingObject USE STEEL_BALL
      physics Physics {
      }
    }
    Solid {
      translation 0 -0.007 -0.03
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry DEF STEEL_BALL Sphere {
            radius 0.007
          }
        }
      ]
      name "solid(3)"
      boundingObject USE STEEL_BALL
      physics Physics {
      }
    }
    DEF left HingeJoint {
      jointParameters HingeJointParameters {
        position 219.71634759566376
      }
      device [
        DEF MotorL RotationalMotor {
          name "MotorR"
        }
      ]
      endPoint Solid {
        translation 0.053499999999999014 0 0
        rotation 0.0969550650940821 -0.09695542123057764 -0.9905550775431942 4.70289564118708
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0.2 0.2 0.4
              roughness 0.8
              metalness 0
            }
            geometry DEF WHEEL Cylinder {
              height 0.005
              radius 0.015
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF right HingeJoint {
      jointParameters HingeJointParameters {
        position 234.65749532460677
      }
      device [
        RotationalMotor {
          name "MotorL"
        }
      ]
      endPoint Solid {
        translation -0.05349999999999976 0 0
        rotation -0.6633814855062703 0.6633839222470024 -0.3461889316423579 3.8081416466606215
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0.2 0.2 0.4
              roughness 0.8
              metalness 0
            }
            geometry DEF WHEEL Cylinder {
              height 0.005
              radius 0.015
            }
          }
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "C:/Users/HP/Desktop/TEMP/nature2.jpeg"
          ]
        }
        roughness 0.2
        metalness 0
      }
      geometry DEF CHASSIS Cylinder {
        height 0.01
        radius 0.05
      }
    }
  ]
  boundingObject USE CHASSIS
  physics Physics {
  }
  controller ""
}
